gammapy/cube/tests/test_psf_map.py

```diff
@@ -182,3 +187,26 @@ def test_psfmap_stacking():
 
 
 # TODO: add a test comparing make_mean_psf and PSFMap.stack for a set of observations in an Observations
+
+
+def test_sample_coord():
+    psf_map = make_test_psfmap(0.1 * u.deg, shape="gauss")
+    coords = MapCoord(
+        {"lon": [0, 0] * u.deg, "lat": [0, 0.5] * u.deg, "energy": [1, 3] * u.TeV},
+        coordsys="GAL",
+    )
+
+    coords_corrected = psf_map.sample_coord(map_coord=coords)
+
+    events = Table()
+    events["LON_TRUE"] = coords_corrected.lon
+    events["LAT_TRUE"] = coords_corrected.lat
+    events["ENERGY_TRUE"] = coords["energy"]
+
+    try:
```

Comment:
You want to establish which frame it is.
So remove try-except, and only keep the assert statement on the frame that you actually get.